repository/model/release.go

```diff
@@ -20,18 +20,41 @@ type Release struct {
 	GitTagName   string    `db:"git_tag_name"`
 	// GithubRepoSlug and GithubOwnerSlug are fetched from the project
 	// and are used to generate the tag URL
-	GithubRepoSlug  sql.NullString `db:"github_repo_slug"`
-	GithubOwnerSlug sql.NullString `db:"github_owner_slug"`
-	CreatedAt       time.Time      `db:"created_at"`
-	UpdatedAt       time.Time      `db:"updated_at"`
+	GithubRepoSlug  sql.NullString      `db:"github_repo_slug"`
+	GithubOwnerSlug sql.NullString      `db:"github_owner_slug"`
+	Attachments     []ReleaseAttachment `db:"attachments"`
+	CreatedAt       time.Time           `db:"created_at"`
+	UpdatedAt       time.Time           `db:"updated_at"`
+}
+
+type ReleaseAttachment struct {
+	ID        uuid.UUID `json:"id"`
+	Name      string    `json:"name"`
+	FilePath  string    `json:"file_path"`
+	CreatedAt time.Time `json:"created_at"`
 }
 
 type gitTagURLGeneratorFunc func(ownerSlug, repoSlug, tag string) (url.URL, error)
+type fileURLGeneratorFunc func(filePath string) (url.URL, error)
 
-func ToSvcRelease(rls Release, urlGenerator gitTagURLGeneratorFunc) (svcmodel.Release, error) {
-	tagURL, err := urlGenerator(rls.GithubOwnerSlug.String, rls.GithubRepoSlug.String, rls.GitTagName)
+func ToSvcRelease(
+	rls Release,
+	tagURLGenerator gitTagURLGeneratorFunc,
+	fileURLGenerator fileURLGeneratorFunc,
+) (svcmodel.Release, error) {
+	tagURL, err := tagURLGenerator(rls.GithubOwnerSlug.String, rls.GithubRepoSlug.String, rls.GitTagName)
 	if err != nil {
-		return svcmodel.Release{}, fmt.Errorf("failed to generate tag URL: %w", err)
+		return svcmodel.Release{}, fmt.Errorf("generating a git tag URL: %w", err)
+	}
+
+	attachments := make([]svcmodel.ReleaseAttachment, 0, len(rls.Attachments))
+	for _, a := range rls.Attachments {
+		u, err := fileURLGenerator(a.FilePath)
```

Comment:
The repository defines the following interface:

```
type fileURLGenerator interface {
    GenerateFileURL(fileObjectKey string) (url.URL, error)
}
```

- A storage client that satisfies this interface is injected into the repository. 
- A repo model function then accepts this URL generator function as a parameter, enabling the generation of file URLs when converting the repository model to the service model.

This approach follows the same principle previously used for generating GitHub URLs.